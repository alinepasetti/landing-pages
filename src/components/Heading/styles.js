import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`font-size: ${theme.fonts.sizes.medium};`,
  medium: (theme) => css`font-size: ${theme.fonts.sizes.large};`,
  large: (theme) => css`font-size: ${theme.fonts.sizes.xlarge};`,
  huge: (theme) => css`
  font-size: ${theme.fonts.sizes.xhuge};
  ${mediaFont(theme)}`,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.fonts.sizes.xlarge};
  }`;

export const Title = styled.h1`
 ${({ theme, lightMode, uppercase, size }) => css`
 color: ${lightMode ? theme.colors.white : theme.colors.secondaryColor};
 text-transform: ${uppercase ? 'uppercase' : 'none'};
 ${titleSize[size](theme)}
 `}
`;
