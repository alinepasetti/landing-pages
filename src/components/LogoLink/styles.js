import styled, { css } from 'styled-components';

export const Container = styled.a` ${({ theme }) => css`
color: black;
text-decoration: none;
display: flex;
align-items: center;


> img {
  max-height: 3.5rem;
}
`}`;
