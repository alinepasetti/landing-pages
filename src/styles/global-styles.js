import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
html {
  /* this allows us to manage rem measure as if it was pixels */
  font-size: 62.5%;
}
body {
  font-size: 1.6rem;
  ${({ theme }) => css`
  background: ${theme.colors.mainColor};
  font-family: ${theme.fonts.default};
`}
}

h1, h2, h3, h4, h5, h6 {
  ${({ theme }) => css`
  font-family: ${theme.fonts.secondary};
  margin: ${({ theme }) => theme.spacings.large} 0;
`}
}

p {
  margin: ${({ theme }) => theme.spacings.medium} 0;
}

ul, ol {
  margin: ${({ theme }) => theme.spacings.medium};
  padding: ${({ theme }) => theme.spacings.medium};
}

a {
  color: ${({ theme }) => theme.colors.secondaryColor};
}
`;
