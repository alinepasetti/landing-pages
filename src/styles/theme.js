export const theme = {
  colors: {
    mainColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },
  fonts: {
    default: '"Open Sans", sans-serif',
    secondary: '"Montserrat", sans-serif;',
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.6rem',
      xlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    medium: '(max-width: 768px)',
  },
  spacings: {
    xsmall: '8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.6rem',
    xlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
};
