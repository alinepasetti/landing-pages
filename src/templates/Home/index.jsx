import { Heading } from '../../components/Heading';
import * as Styled from './styles';

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper background={'purple'}>
        <Heading>Hello</Heading>
      </Styled.Wrapper>
    </div>
  );
}

export default Home;
