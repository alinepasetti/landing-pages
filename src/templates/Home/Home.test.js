import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import Home from '.';
import { theme } from '../../styles/theme';

test('renders the element with the expected style', () => {
  renderTheme(<Home />);

  const headingContainer = screen.getByRole('heading', {
    name: 'Hello',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: 'purple',
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'purple');
});
