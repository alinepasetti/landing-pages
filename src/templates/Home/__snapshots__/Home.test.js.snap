// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the element with the expected style 1`] = `
.c0 {
  background: purple;
  color: red;
}

<div
  class="c0"
>
  <h1>
    Hello
  </h1>
</div>
`;
